### ISSUE TITLE
losetup is broken because of the /dev/loop dir created by udev rule

### ISSUE THREAD

**Issue by montag451** [systemd/systemd#28475]
### systemd version the issue has been seen with

254

### Used distribution

Debian sid

### Linux kernel version used

6.3.0-2-amd64

### CPU architectures issue was seen on

x86_64

### Component

udev rule files

### Expected behaviour you didn't see

After having associated a file with a loop device, `losetup -a` should list the loop device and its backing file.

### Unexpected behaviour you saw

Since 5ac52d1f7b7cd11cad8b5c2e9812d7ee7560a517, `losetup -a` doesn't print anything and other `losetup` commands are broken (e.g `losetup --find --nooverlap`, `losetup -D`). The reason why it doesn't work anymore is because `losetup` expects to find the loop devices in a `/dev/loop` directory when such a directory exists.

### Steps to reproduce the problem

``` sh
losetup --find /path/to/a/file
losetup -a
```

**Comment by bluca** [comment]
@karelzak wasn't losetup updated for this?

**Comment by gdamjan** [comment]
as of
```
losetup from util-linux 2.39.1
```
it has the issue still

**Comment by karelzak** [comment]
> @karelzak wasn't losetup updated for this?

Nope, if I remember correctly (well, memory is not my strongest skill ;-)), this is the first time I've seen /dev/loop/ used for symlinks. Frankly, it's a bad idea because, in the last decades, some distros have used the directory for loop devices, and losetup has supported this since 2011 (perhaps even earlier).

It would be better to use /dev/disk-by/ for symlinks to ensure safety and backward compatibility. The udev rule should be modified accordingly

**Comment by karelzak** [comment]
I mean "/dev/disk/by-backingfile/" for example.

**Comment by karelzak** [comment]
Now I remember: https://github.com/util-linux/util-linux/issues/2106

**Comment by bluca** [comment]
See, I thought there was something :-) So is that change in util-linux enough? Or do we need to do something else here?

**Comment by karelzak** [comment]
The losetup change is not merged in the master branch yet (my mistake, it is still in my private "next" branch due to the previous feature freeze).

Unfortunately, it seems we need more changes than we have planned with Lennart; the new option --loop-ref will not fix the problem. 

The losetup really doesn't like it if /dev/loop/ does not contain dev nodes. I'll try to fix it for v2.39.2, and  --loop-ref will be in v2.40.

@montag451, please fill a new issue for util-linux (https://github.com/util-linux/util-linux/issues/) and refer to this discussion? Thanks!

**Comment by bluca** [comment]
In the meanwhile, I guess we should revert and avoid releasing with this for now?
